pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -105,9 +105,12 @@ def advance(  # type: ignore[override]
         void(dl_max_batches)
 
         assert self._dataloader_iter is not None
-        batch, self.batch_progress.is_last_batch = next(self._dataloader_iter)
-        if batch is None:
-            raise StopIteration
+        if not isinstance(data_fetcher, DataLoaderIterDataFetcher):
+            batch, self.batch_progress.is_last_batch = next(self._dataloader_iter)
```

Comment:
It would be better to track the batch_idx from the DataFetcher automatically as I did with the DataLoaderIterDataFetcher

Reply:
@ananthsub Can we prevent this check. This would make the code less readable and this was already supported in the past. I don't believe the `training_step` API should change. Batch would just be renamed dataloader_iter, but the batch_idx should still be the exact number of batches fetched by the user.

Reply:
As done there: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/fetching.py#L326

Reply:
What Ananth is adding here matches exactly what we have in the `TrainingEpochLoop`